be/test/io/s3_input_stream_test.cpp

```diff
@@ -138,6 +138,15 @@ TEST_F(S3InputStreamTest, test_read) {
     ASSERT_EQ(10, *f->position());
 }
 
+TEST_F(S3InputStreamTest, test_not_found) {
+    auto f = std::make_unique<S3InputStream>(g_s3client, s_bucket_name, "key_not_found");
+    char buf[6];
+    auto r = f->read(buf, sizeof(buf));
+    EXPECT_TRUE(r.status().message().find("SdkResponseCode=404") != std::string::npos);
+    // ErrorCode 16 means RESOURCE_NOT_FOUND
+    EXPECT_TRUE(r.status().message().find("SdkErrorType=16") != std::string::npos);
+}
+
```

Comment:
```
inline Status make_error_status(const Aws::S3::S3Error& error) {
    return Status::IOError(fmt::format(
            "BE access S3 file failed, SdkResponseCode={}, SdkErrorType={}, SdkErrorMessage={}",
            static_cast<int>(error.GetResponseCode()), static_cast<int>(error.GetErrorType()), error.GetMessage()));
}
```

It is too vague to mapping all s3 errors to IOError. Please open an issue to improve this.

Reply:
done, https://github.com/StarRocks/starrocks/issues/52602